src/backend/BackendFactory.h

```diff
@@ -57,6 +57,10 @@ make_Backend(boost::asio::io_context& ioc, clio::Config const& config)
 
     log.info() << "Constructed BackendInterface Successfully";
 
+    log.info() << __func__ << ": Running migrations";
+    backend->runMigrations();
```

Comment:
that's exactly why i put it in the factory. otherwise were you thinking of putting it in the BackendInterface constructor? In that case, isn't it called before the child constructor? that wouldn't work.

Reply:
I think we should leave migrations entirely to each implementation instead of artificially trying to go thru BackendInterface. 
Each DB will have completely different migration system, schema and everything else specific to the DB. Furthermore we support Cassandra today with Schema `1`, tomorrow it will be `2` and next year we support another Database entirely which will start with schema `0` (which already maps to schema `2` of Cassandra at that point). 
In any case - factory is never the right place for any logic other than creation of objects.

Reply:
move this into CassandraBackend somewhere